lib/fs.js

```diff
@@ -688,14 +688,19 @@ fs.renameSync = function(oldPath, newPath) {
 };
 
 fs.truncate = function(path, len, callback) {
-  if (typeof path === 'number') {
+  if (Number.isInteger(path) && path >= 0)
```

Comment:
I just followed the logic paths for current and this patch, and there seemed to be a discrepancy.